src/imperative/imperative.cc

```diff
@@ -367,6 +369,10 @@ void Imperative::RecordDeferredCompute(nnvm::NodeAttrs&& attrs,
     node_count_++;
   }
 
+  if (get_opt_constraints() != OptConstraint::None) {
+    node->attrs.dict[OPT_CONSTRAINT_ATTR] = std::to_string(static_cast<int>(get_opt_constraints()));
```

Comment:
```suggestion
    node->attrs.dict[OPT_CONSTRAINT_ATTR] = std::to_string(static_cast<unsigned int>(get_opt_constraints()));
```
or maybe it is worth to overload operator << to make it human readable